### PR TITLE
Output directory adapter lookup

### PR DESCRIPTION
**Pull request by singhsarab**
Changes for picking up adapters from the output directory in case testadapterpath is not given.

### CODE REVIEW

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs [hunk 1]
```diff
@@ -39,10 +41,12 @@ public class DefaultTestHostManager : ITestRuntimeProvider
 
         private const string DefaultTestHostUri = "HostProvider://DefaultTestHost";
         private const string DefaultTestHostFriendlyName = "DefaultTestHost";
+        private const string TestAdapterRegexPattern = @"TestAdapter.dll";
```
Comment (codito) [inline src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: We moved away from regex to simple string match IIRC

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs [hunk 2]
```diff
@@ -176,6 +183,11 @@ public TestHostConnectionInfo GetTestHostConnectionInfo()
         /// <inheritdoc/>
         public IEnumerable<string> GetTestPlatformExtensions(IEnumerable<string> sources, IEnumerable<string> extensions)
         {
+            if (sources != null && sources.Any() && this.projectOutputExtensionsRequired)
```
Comment (codito) [inline src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: Does this work for not isolation scenario?

Comment (singhsarab) [reply src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: Yes.

File: test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs [hunk 3]
```diff
@@ -30,6 +30,19 @@ public void RunMultipleTestAssemblies(string runnerFramework, string targetFrame
         [CustomDataTestMethod]
         [NETFullTargetFramework]
         [NETCORETargetFramework]
+        public void RunMultipleTestAssembliesWithoutTestAdapterPath(string runnerFramework, string targetFramework, string targetRuntime)
+        {
+            AcceptanceTestBase.SetTestEnvironment(this.testEnvironment, runnerFramework, targetFramework, targetRuntime);
+
+            var assemblyPaths =
+                this.BuildMultipleAssemblyPath("SimpleTestProject.dll", "SimpleTestProject2.dll").Trim('\"');
```
Comment (smadala) [inline test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs]: Pass xunit dll as second dll.

**Review by smadala — APPROVED** [review]

**Comment by smadala** [comment]
Fixes: https://github.com/Microsoft/vstest/issues/251

File: test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs (hunk 3 above)
Comment (smadala) [reply test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs]: Remove /testAdapterPath:"$testAdapterPath" in test.ps1.

**Review by mayankbansal018 — APPROVED** [review]

**Pull request merged**